docs/recipes/ui/add-routing.md

```diff
@@ -0,0 +1,203 @@
+              Todos = []
+              Input = "" }
+        ...
+        model, cmd
+    ```
+=== "Diff"
+    ```diff
+      let init () : Model * Cmd<Msg> =
+    +     let page = Router.currentUrl () |> parseUrl
+    +
+    -      let model = { Todos = []; Input = "" }
+    +      let model =
+    +        { CurrentPage = page
+    +         Todos = []
+    +         Input = "" }
+          ...
+          model, cmd
+    ```
+## 5. Updating the URL
+
+Add an action to handle navigation.
+
+To the `Msg` type, add a `PageChanged` case of `Page`
+
+=== "Code"
+    ```fsharp
+    type Msg =
+        ...
+        | PageChanged of Page
+    ```
+=== "diff"
+    ```.diff 
+     type Msg =
+         ...
+    +    | PageChanged of Page
+    ```
+
+Add the `PageChanged` update action
+
+=== "Code"
+    ```fsharp
+    let update (msg: Msg) (model: Model) : Model * Cmd<Msg> =
+        match msg with
+        ...
+        | PageChanged page -> { model with CurrentPage = page }, Cmd.none
+    ```
+=== "Diff"
+    ```.diff
+      let update (msg: Msg) (model: Model) : Model * Cmd<Msg> =
+          match msg with
+          ...
+    +     | PageChanged page -> { model with CurrentPage = page }, Cmd.none
+    ```
+
+## 6. Displaying the correct content
+
+Rename the `view` function to `todoView`
+
+=== "Code"
+    ```fsharp
+    let todoView (model: Model) (dispatch: Msg -> unit) =
+        Bulma.hero [
+        ...
+        ]
+    ```
+=== "Diff"
+    ```.diff
+    - let view (model: Model) (dispatch: Msg -> unit) =
+    + let todoView (model: Model) (dispatch: Msg -> unit) =
+          Bulma.hero [
+          ...
+          ]
+    ```
+
+Add a new view function, that returns the appropriate page
+
+```fsharp
+let view (model: Model) (dispatch: Msg -> unit) =
+    match model.CurrentPage with
+    | TodoList -> todoView model dispatch
+    | NotFound -> Bulma.box "Page not found"
+    
+```
+
+!!! info "Adding UI elements to every page of the website"
+    In this recipe, we moved all the page content to the `todoView`, but you don't have to. You can add UI you want to display on every page of the application to the `view` function.
+
+## 7. Adding the React router to the view
+
+Add the `React.Router` element as the outermost element of the view. Dispatch the PageChanged event on `onUrlChanged`
+
+=== "Code"
+    ```fsharp
+    let view (model: Model) (dispatch: Msg -> unit) =
+        React.router [
+            router.onUrlChanged (parseUrl >> PageChanged >> dispatch)
+            router.children [
+                match model.CurrentPage with
+                ...
+            ]
+        ]
+    ```
+=== "Diff"
+    ```.diff
+      let view (model: Model) (dispatch: Msg -> unit) =
+    +     React.router [
+    +         router.onUrlChanged (parseUrl >> PageChanged >> dispatch)
+              router.children [
+                  match model.CurrentPage with
+                  ...
+              ]
+          ]
+    ```
+
+## 9.  Try it out 
```

Comment:
The suggestions for the title/heading have got me thinking - part of the benefit of this approach is that state is never lost. This means that you can navigate away from a route and then back and the state is still there. For example, you can enter some text in the todo text input, browse to localhost:8080, then back to localhost:8080/#/todo and the text will still be in the input. I think that it would be worth explicitly mentioning this in the try it out section.

I would also mention it in the introduction - it's a good reason why someone might choose this recipe over an alternative.

Reply:
Added that